src/main/java/configurationslicing/maven/MavenDisableArtifact.java

```diff
@@ -0,0 +1,49 @@
+package configurationslicing.maven;
+
+import configurationslicing.BooleanSlicer;
+import hudson.Extension;
+import hudson.maven.MavenModuleSet;
+import jenkins.model.Jenkins;
+
+import java.util.List;
+
+@Extension
+public class MavenDisableArtifact extends BooleanSlicer<MavenModuleSet> {
+
+    public MavenDisableArtifact() {
+        super(new MavenDisableArtifactSlicerSpec());
+    }
+    
+    public static class MavenDisableArtifactSlicerSpec implements BooleanSlicerSpec<MavenModuleSet> {
+
+        public String getName() {
+            return "Maven Disable automatic artifact archiving";
```

Comment:
I created an issue for that to review all the code of the plugin (see https://issues.jenkins-ci.org/browse/JENKINS-44363)